dotm_search.py

```diff
@@ -4,4 +4,69 @@
 within the 'word/document.xml' section of a MSWord .dotm file.
 """
 
-# Your awesome code begins here!
+import os
+import argparse
+import zipfile
+
+DOC_FILENAME = 'word/document.xml'
+
+def find_dotm(path_to_search, text_to_search):
+    """Iterate through dotm files, looking for text"""
+    file_list = os.listdir(path_to_search)
+
+    files_searched = 0
+    files_matched = 0
+
+    for file in file_list:
+        if not file.endswith('.dotm'):
+            continue
+        full_path = os.path.join(path_to_search, file)
+        if not zipfile.is_zipfile(full_path):
+            continue
+        files_searched += 1
+
+        with zipfile.ZipFile(full_path) as z:
+            if DOC_FILENAME in z.namelist():
+                with z.open(DOC_FILENAME) as doc:
+                    for line in doc:
+                        text_location = line.find(text_to_search)
+                        if text_location >= 0:
+                            files_matched += 1
+                            if text_location - 40 >= 0:
+                                left_location = text_location - 40
+                            else:
+                                left_location = 0
+                            if text_location + 41 <= len(line):
+                                right_location = text_location + 41
+                            else:
+                                right_location = len(line)
```

Comment:
This is good, being careful to not index beyond the beginning or end of the line to extract the +/- 40 chars of context.  Could you condense this with application of a min() or max() function e.g. `left_location = max(0, text_location-40)` ?